### PR TITLE
feat: add architecture for record linkage 

### PR DESCRIPTION
**Pull request by alejandrosame**

### CODE REVIEW

**Comment by alejandrosame** [comment]
I took #84 as starting point. The rough roadmap is:

- provide a functional manual triage view
- show candidate linked records in the triage view
- feed candidates from recordlinkage heuristics.

File: src/website/shared/migrations/0025_affectedproduct_search_vector_and_more.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            model_name='cpe',
+            trigger=pgtrigger.compiler.Trigger(name='cpe_search_vector_idx', sql=pgtrigger.compiler.UpsertTriggerSql(execute='tsvector_update_trigger("search_vector", "pg_catalog.english", "name")', func='', hash='5831e70cff7886047233ef2d3b870a320fa5d81d', operation='INSERT OR UPDATE OF "name"', pgid='pgtrigger_cpe_search_vector_idx_80861', table='shared_cpe', when='BEFORE')),
+        ),
+        pgtrigger.migrations.AddTrigger(
+            model_name='description',
+            trigger=pgtrigger.compiler.Trigger(name='description_search_vector_idx', sql=pgtrigger.compiler.UpsertTriggerSql(execute='tsvector_update_trigger("search_vector", "pg_catalog.english", "value")', func='', hash='5b6196ba33f28ffdb9dcfff123f0ea2fe94588fd', operation='INSERT OR UPDATE OF "value"', pgid='pgtrigger_description_search_vector_idx_ce47a', table='shared_description', when='BEFORE')),
+        ),
+        # Make sure that the VectorFields are populated when there's data before this migration runs
+        migrations.RunPython(
+            update_search_vectors, reverse_code=migrations.RunPython.noop
```
Comment (RaitoBezarius) [inline src/website/shared/migrations/0025_affectedproduct_search_vector_and_more.py:80]: The reverse code should exist.

Comment (alejandrosame) [reply src/website/shared/migrations/0025_affectedproduct_search_vector_and_more.py:80]: Unless I'm misunderstanding something, the reverse code of updating the search vector fields for the first time after creating the fields is really a noop. 

If this whole migration is reverted, those fields will be dropped.

Comment (RaitoBezarius) [reply src/website/shared/migrations/0025_affectedproduct_search_vector_and_more.py:80]: Urgh, you're right! Apologies for the noise.

Comment (alejandrosame) [reply src/website/shared/migrations/0025_affectedproduct_search_vector_and_more.py:80]: No problem!

**Comment by fricklerhandwerk** [comment]
@alejandrosame can we salvage anything from here except for more self-descriptive names? Seems like the business logic was ported a while ago.

**Comment by alejandrosame** [comment]
V1 already implemented in https://github.com/Nix-Security-WG/nix-security-tracker/pull/254

**Comment by alejandrosame** [comment]
> @alejandrosame can we salvage anything from here except for more self-descriptive names? Seems like the business logic was ported a while ago.

I didn't noticed this question before. I don't think so. The original discussions were nudging the implementation to use or at least consider the recordlinkage python library, so if anything https://github.com/Nix-Security-WG/nix-security-tracker/pull/254 should have taken into account these changes.

Since clearly code wasn't reused there, anything going forward should just go from there. It's not like the vision was completely set before attempting this anyway.

**Pull request closed without merging**